mcs/class/System/System.Diagnostics/PerformanceCounterCategory.cs

```diff
@@ -40,31 +40,41 @@ public sealed class PerformanceCounterCategory
 		private string machineName;
 		private PerformanceCounterCategoryType type = PerformanceCounterCategoryType.Unknown;
 
+// FIXME Span<char>?
+
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
-		static extern bool CategoryDelete (string name);
+		unsafe static extern bool CategoryDelete (char* category, int category_length);
 
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
-		static extern string CategoryHelpInternal (string category, string machine);
+		unsafe static extern string CategoryHelpInternal (char* category, int category_length);
 
 		/* this icall allows a null counter and it will just search for the category */
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
-		static extern bool CounterCategoryExists (string counter, string category, string machine);
+		unsafe static extern bool CounterCategoryExists (char* counter,
+			int counter_length, char* category, int category_length);
 
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
-		static extern bool Create (string categoryName, string categoryHelp,
+		unsafe static extern bool Create (char* categoryName, int categoryName_length,
+			char* categoryHelp, int categoryHelp_length,
 			PerformanceCounterCategoryType categoryType, CounterCreationData[] items);
 
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
-		static extern int InstanceExistsInternal (string instance, string category, string machine);
+		unsafe static extern int InstanceExistsInternal (char* instance,
+			int instance_length, char* category, int category_length);
 
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
-		static extern string[] GetCategoryNames (string machine);
+		unsafe static extern string[] GetCategoryNames ();
 
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
-		static extern string[] GetCounterNames (string category, string machine);
+		unsafe static extern string[] GetCounterNames (char* category, int category_length);
 
 		[MethodImplAttribute (MethodImplOptions.InternalCall)]
-		static extern string[] GetInstanceNames (string category, string machine);
+		unsafe static extern string[] GetInstanceNames (char* category, int category_length);
+
+		static int StringLength (string a)
```

Comment:
All the `categoryName` and `machineName` are already guaranteed to be non-`null`, so all these null checks are redundant. All the calls to `StringLength(categoryName)` can safely be replaced by `categoryName.Length` for example.

Reply:
Ok, I didn't realize that.